graviti/schema/catalog_to_schema.py

```diff
@@ -0,0 +1,268 @@
+}
+
+_JSON_TYPE_TO_PORTEX_TYPE = {
+    "array": "array",
+    "boolean": "boolean",
+    "integer": "int32",
+    "number": "float32",
+    "string": "string",
+}
+
+_LABEL_PROCESSORS: Dict[str, Callable[..., Any]] = {}
+
+
+def _get_category_params(subcatalog: Union[CategoriesMixin, MaskCategoriesMixin]) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "categories"):
+        return {}
+
+    category_params: Dict[str, Any] = {}
+    if isinstance(subcatalog, MaskCategoriesMixin):
+        category_params["categoryIds"] = subcatalog.get_index_to_category()
+    else:
+        category_params["categories"] = list(subcatalog.categories.keys())
+
+    if getattr(subcatalog, "category_delimiter", None):
+        category_params["categoryDelimiter"] = subcatalog.category_delimiter
+
+    return category_params
+
+
+def _get_attribute_params(subcatalog: AttributesMixin) -> Dict[str, Any]:
+    if not hasattr(subcatalog, "attributes"):
+        return {}
+
+    attribute_params: List[Dict[str, Any]] = []
+    for attribute in subcatalog.attributes:
+        field: Dict[str, Any] = {"name": attribute.name}
+        if hasattr(attribute, "enum"):
+            field["type"] = "enum"
+            field["values"] = attribute.enum
+        else:
+            attribute_type = (
+                reduce(lambda a, b: a or b, attribute.type)
+                if isinstance(attribute.type, list)
+                else attribute.type
+            )
+            field["type"] = _JSON_TYPE_TO_PORTEX_TYPE[attribute_type]
+
+        if hasattr(attribute, "minimum"):
+            field["minimum"] = attribute.minimum
+        if hasattr(attribute, "maximum"):
+            field["maximum"] = attribute.maximum
+        attribute_params.append(field)
+    return {"attributes": attribute_params}
+
+
+def _get_category_and_attribute_params(subcatalog: Any) -> Dict[str, Any]:
+    params = _get_category_params(subcatalog)
+    params.update(_get_attribute_params(subcatalog))
+    return params
+
+
+def _get_dtype(value: float, default_type: str = "int") -> Dict[str, str]:
+    dtype = type(value).__name__
+    return {"dtype": f"{dtype}32"} if dtype != default_type else {}
+
+
+class _ConverterClassification:  # pylint: disable=too-few-public-methods
+    _PROCESSORS = (
+        ("category", "label.Category", _get_category_params),
+        ("attribute", "label.Attribute", _get_attribute_params),
+    )
+
+    def __init__(self, label_type: str) -> None:
+        _LABEL_PROCESSORS[label_type] = self
+
+    def __call__(self, subcatalog: ClassificationSubcatalog) -> Iterator[Dict[str, Any]]:
+        for field_name, field_type, processor in self._PROCESSORS:
+            params = processor(subcatalog)  # type: ignore[operator]
+            if params:
+                field = {"name": field_name, "type": field_type}
+                field.update(params)
+                yield field
+
+
+class _ConverterBase:  # pylint: disable=too-few-public-methods
```

Comment:
wield, why `ConverterClassification` is not a subclass of `ConverterBase`?

Reply:
Since the input arguments and the output type of classification converter are different from other converters.